packages/cli/src/commands/validator/status.ts

```diff
@@ -0,0 +1,117 @@
+      description: 'address of the validator to check if elected and validating',
+    }),
+    validator: Flags.address({
+      description: 'address of the signer to check if elected and validating',
+    }),
+    lookback: flags.integer({
+      description: 'how many blocks to look back for signer activity',
+      default: 50,
+    }),
+  }
+
+  static examples = [
+    'status --validator 0x5409ED021D9299bf6814279A6A1411A7e866A631',
+    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c',
+    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c --lookback 100',
+  ]
+
+  requireSynced = true
+
+  async run() {
+    const { flags } = this.parse(ValidatorOnline)
+
+    if (!flags.validator && !flags.signer) {
+      this.error('Either validator or signer must be specified')
+    }
+
+    // Determine the validator signer to check.
+    let signer: Address
+    if (flags.validator) {
+      const accounts = await this.kit.contracts.getAccounts()
+      signer = await accounts.getValidatorSigner(flags.validator)
+      console.info(`Identified ${signer} as the authorized validator signer`)
+
+      // If the user specified both, verify that it matches what is stored on chain.
+      if (flags.signer && signer.toLowerCase() !== flags.signer.toLowerCase()) {
+        this.error(`${flags.signer} is not the authorized signer for ${flags.validator}`)
+      }
+    } else {
+      signer = flags.signer!
+    }
+
+    // Use redundant checks to help the user diagnose issues.
+    await newCheckBuilder(this, signer)
+      .canSignValidatorTxs()
+      .signerMeetsValidatorBalanceRequirements()
+      .signerAccountIsValidator()
+      .runChecks()
+
+    // Determine if the signer is elected, and get their index in the validator set.
+    const election = await this.kit.contracts.getElection()
+    const signers = await election.getCurrentValidatorSigners()
+    const signerIndex = signers.map((a) => a.toLowerCase()).indexOf(signer.toLowerCase())
+    if (signerIndex < 0) {
+      // Determine whether the signer will be elected at the next epoch to provide a helpful error.
+      const frontrunners = await election.electValidatorSigners()
+      if (frontrunners.map((a) => a.toLowerCase()).indexOf(signer.toLowerCase()) >= 0) {
+        this.error(
+          `Signer ${signer} is not elected for this epoch, but is currently winning in the upcoming election. Wait for the next epoch.`
+        )
+      } else {
+        this.error(
+          `Signer ${signer} is not elected for this epoch, and is not currently winning the upcoming election.`
+        )
+      }
+    }
+    console.info('Signer has been elected for this epoch')
+
+    if (flags.lookback <= 0) {
+      return
+    }
+
+    // Retrieve blocks to examine for the singers signature.
+    cli.action.start(`Retreiving the last ${flags.lookback} blocks`)
+    const latest = await this.web3.eth.getBlock('latest')
+    const blocks = await Promise.all(
```

Comment:
Probably not a problem here, but in general it's better to use "bounded" concurrency. That is, have control on how many things you are doing at the same time, since resources are always limited (be it processor, bandwith, whatever). In this case, probably doesn't make sense. 

If not, there's a `concurrentMap` in @celo/utils